entity-framework/core/providers/cosmos/querying.md

```diff
@@ -0,0 +1,312 @@
+
+Rather than referencing partition key properties in a `Where` operator, you can explicitly specify them by using the <xref:Microsoft.EntityFrameworkCore.CosmosQueryableExtensions.WithPartitionKey%2A> operator:
+
+```c#
+var sessions = await context.Sessions
+    .WithPartitionKey(tenantId, userId)
+    .Where(e => e.SessionId > 0 && e.Username.Contains("a"))
+    .ToListAsync();
+```
+
+This executes in the same way as the above query, and can be preferable if you want to make partition keys more explicit in your queries. Using <xref:Microsoft.EntityFrameworkCore.CosmosQueryableExtensions.WithPartitionKey%2A> may be necessary in versions of EF prior to 9.0 - keep an eye on the logs to ensure that queries are using partition keys as expected.
+
+## Point reads
+
+While Azure Cosmos DB allows for powerful querying via SQL, such queries can be quite expensive. Cosmos DB also supports _point reads_, which can be used when both the `id` property and the entire partition key are known. Such point reads directly identify a specific document in a specific partition, and execute extremely efficiently and with reduced costs. If at all possible, it's worth designing your system in a way which leverages point reads as much as possible. To read more, see the [Cosmos DB documentation](/azure/cosmos-db/nosql/how-to-dotnet-read-item).
+
+In the previous section, we saw EF identifying and extracting partition key comparisons from the `Where` clause for more efficient querying, restricting processing only to the relevant partitions. It's possible to go a step further, and provide the `id` property in the query as well. Let's examine the following query:
+
+```c#
+var session = await context.Sessions.SingleAsync(
+    e => e.Id == someId
+         && e.TenantId == tenantId
+         && e.UserId == userId
+         && e.SessionId == sessionId);
+```
+
+In this query, a value for the `Id` property is provided (which is mapped to the Cosmos DB `id` property), as well as values for all the partition key properties. Furthermore, there are no additional components to the query. When all these conditions are met, EF is able to execute the query as a point read:
+
+```console
+Executed ReadItem (46 ms, 1 RU) ActivityId='d7391311-2266-4811-ae2d-535904c42c43', Container='test', Id='9', Partition='["Microsoft","99a410d7-e467-4cc5-92de-148f3fc53f4c",10.0]'
+```
+
+Note the `ReadItem`, which indicates that the query was executed as an efficient point read - no SQL query is involved.
+
+Note that as with partition key extraction, significant improvements have been made to this mechanism in EF 9.0; older versions do not reliably detect and use point reads.
+
+## Pagination
+
+> [!NOTE]
+> This feature was introduced in EF Core 9.0.
```

Comment:
Should mention that it's currently experimental as we want to hear feedback before we finalize the API shape